### DISCUSSION TITLE
Encrypting npmToken in self hosted

### DISCUSSION THREAD

**Discussion by haldunanil** [renovatebot/renovate#8037] [Request Help] [answered]
I'm trying to set up an encrypted `npmToken` in the self hosted version of the bot, but I'm having some difficulty actually encrypting it. 

I followed the instructions [here](https://docs.renovatebot.com/self-hosted-configuration/#privatekey), specifically the bit that goes:

> To create the key pair with openssl use the following commands:
> 
> `openssl genrsa -out rsa_priv.pem 4096` for generating the private key
> `openssl rsa -pubout -in rsa_priv.pem -out rsa_pub.pem` for extracting the public key

and I was able to do those no problem. However, there's no clear guide on how to actually encrypt an `npmToken` (or anything for that matter) once those private and public keys are set up. https://renovatebot.com/encrypt doesn't work since the public keys are different. 

How do I encrypt the `npmToken` in a way that the private key specified by `privateKey`/`privateKeyPath` can then decrypt? Alternatively, is there a way I can do this in the `config.json` file in the self hosted config?

**Comment by viceice** [comment, marked as answer]
![image](https://user-images.githubusercontent.com/1798109/102311297-e7a06b00-3f6c-11eb-9d27-4e0c2d53d33d.png)
 Did you read this? You can simply download that page, replace the public key and open in a browser to use a simple ui.

**Reply by rarkins** [reply]
Yes, edit the source of that HTML page to insert your own public key

**Reply by haldunanil** [reply]
Ah, I see now. My bad, thanks!
